config/config-api/src/main/java/com/thoughtworks/go/config/PipelineConfig.java

```diff
@@ -367,6 +367,15 @@ public boolean isConfigDefinedRemotely() {
         return this.origin instanceof RepoConfigOrigin;
     }
 
+    public boolean hasSameConfigOrigin(PipelineConfig other) {
+        if (!(isConfigDefinedRemotely()))
+            return false;
+
+        RepoConfigOrigin otherOrigin = (RepoConfigOrigin) other.getOrigin();
+        RepoConfigOrigin repoConfigOrigin = (RepoConfigOrigin) this.origin;
```

Comment:
The cast is unnecessary and may fail when origin on `other` is not remote.
All implementations of ConfigOrigin can handle equality. You can just use
```
this.origin.equals(otherOrigin)
```

Also, name of method is confusing. `hasSameConfigOrigin` suggests that origin is of any type. But you actually meant `hasSameRemoteConfigOrigin`